test/tests/LibStuffTest.cpp

```diff
@@ -633,4 +637,39 @@ struct LibStuff : tpunit::TestFixture {
         ASSERT_EQUAL(SFirstOfMonth(timeStamp4, -13), "2019-06-01");
         ASSERT_EQUAL(SFirstOfMonth(timeStamp4, -25), "2018-06-01");
     }
+
+    void SQResultTest() {
+        SQLite db(":memory:", 1000, 1000, 1);
+        db.beginTransaction(SQLite::TRANSACTION_TYPE::EXCLUSIVE);
+        db.write("CREATE TABLE testTable(id INTEGER PRIMARY KEY, name STRING, value STRING, created DATE);");
+        db.write ("INSERT INTO testTable VALUES(1, 'name1', 'value1', '2023-12-20');");
+        db.write ("INSERT INTO testTable VALUES(2, 'name2', 'value2', '2023-12-21');");
+        db.write ("INSERT INTO testTable VALUES(3, 'name3', 'value3', '2023-12-22');");
+        db.prepare();
+        db.commit();
+
+        db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED);
+        SQResult result;
+        db.read("SELECT name, value FROM testTable ORDER BY id;", result);
+        db.rollback();
+
+        // All our names make sense?
+        ASSERT_EQUAL(result.cell(0, "name"), "name1");
```

Comment:
Hmmmm would be nice if the way to access this was: `result[0].cell("name")`... is there a way to do that that does not require rewriting everything we already have?

Reply:
I would have done that if I could have easily. The issue is that `result[0]` is a simple `vector<string>`. We would have to change that to some other type that kept the names. We could feasibly do that by inheriting from `vector<string>`, i.e., something like:
```
class SQResultRow : public vector<string> {
};
```

And that might mostly work in most places, but the titles themselves are not a part of the row. We'd either need to copy the titles into each row, or we'd need to make the rows dependent upon the original `SQResult`, which is probably OK, but this would be significantly more intrusive to existing code.

I guess I could try that approach and see how many things break.

Reply:
Let me just try it real quick.

Reply:
I think I got it to work, just need to run the auth tests and verify.